crates/jstz_cli/src/deploy.rs

```diff
@@ -7,21 +7,35 @@ use jstz_proto::{
 use log::{debug, info};
 
 use crate::{
+    account,
     config::{Config, NetworkName, SmartFunction},
     error::{bail, bail_user_error, user_error, Result},
+    term::styles,
     utils::read_file_or_input_or_piped,
 };
 
 pub async fn exec(
-    code: Option<String>,
+    code_op: Option<String>,
     balance: u64,
     name: Option<String>,
     network: Option<NetworkName>,
 ) -> Result<()> {
     let mut cfg = Config::load()?;
+    // Load sandbox if the selected network is Dev and sandbox is not already loaded
+    if cfg.network_name(&network)? == NetworkName::Dev && cfg.sandbox.is_none() {
+        bail_user_error!(
+            "No sandbox is currently running. Run {} for an interactive session or {} to run in the background.",
+            styles::command("jstz sandbox start"),
+            styles::command("jstz sandbox start --detach")
+        );
+    }
 
-    let (_, user) = cfg.accounts.current_user().ok_or(user_error!(
-        "You are not logged in. Please run `jstz login`."
+    // Get the current user and check if we are logged in
+    account::login_quick(&mut cfg)?;
+    cfg.reload()?;
+    let (user_name, user) = cfg.accounts.current_user().ok_or(user_error!(
+        "Failed to setup the account. Please try `{}`.",
```

Comment:
Consistency with other messages

```suggestion
        "Failed to setup the account. Please run`{}`.",
```